clients/lsp-rust.el

```diff
@@ -1094,6 +1095,22 @@ open in a new window."
         (lsp--warn "Couldn't find a Cargo.toml file or your version of rust-analyzer doesn't support this extension"))
     (lsp--error "OpenCargoToml is an extension available only with rust-analyzer")))
 
+(defun lsp-rust-analyzer-open-external-docs ()
+  "Open a URL for documentation related to the current TextDocumentPosition.
+
+Rust-Analyzer LSP protocol documented here
+https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/lsp-extensions.md#open-external-documentation"
+  (interactive)
+  (-if-let (workspace (lsp-find-workspace 'rust-analyzer))
```

Comment:
can you remove that -if-let? The if-let in `lsp-rust-analyzer-open-cargo-toml` because you can call this function in `dired` buffer for example and it will work. In this case, there is no need to look for workspace - just execute the function using the current context.  You may follow `lsp-rust-analyzer-move-item` as an example.